packages/prettier-plugin-java/src/printers/expressions.ts

```diff
@@ -324,10 +321,50 @@ export class ExpressionsPrettierVisitor extends BaseCstPrettierPrinter {
     params?: { addParenthesisToWrapStatement?: boolean }
   ) {
     const countMethodInvocation = isUniqueMethodInvocation(ctx.primarySuffix);
+    const newExpression =
+      ctx.primaryPrefix[0].children.newExpression?.[0].children;
+    const arrayCreationExpression =
+      newExpression?.arrayCreationExpression?.[0].children;
+    const classInstanceCreationExpression =
+      newExpression?.unqualifiedClassInstanceCreationExpression?.[0].children;
+    const isBreakableNewExpression =
+      countMethodInvocation <= 1 &&
+      [
+        arrayCreationExpression?.classOrInterfaceType?.[0].children.classType[0]
+          .children.typeArguments,
+        arrayCreationExpression?.arrayCreationExplicitInitSuffix?.[0].children
+          .arrayInitializer[0].children.variableInitializerList,
+        classInstanceCreationExpression?.classOrInterfaceTypeToInstantiate[0]
+          .children.typeArgumentsOrDiamond?.[0].children.typeArguments,
+        classInstanceCreationExpression?.argumentList
+      ].some(breakablePart => breakablePart);
+    const fqnOrRefType =
+      ctx.primaryPrefix[0].children.fqnOrRefType?.[0].children;
+    const fqnOrRefTypeParts = [
+      fqnOrRefType?.fqnOrRefTypePartFirst[0],
+      ...(fqnOrRefType?.fqnOrRefTypePartRest ?? [])
+    ];
+    const nextToLastIdentifier =
+      fqnOrRefTypeParts[fqnOrRefTypeParts.length - 2]?.children
+        .fqnOrRefTypePartCommon[0].children.Identifier?.[0].image;
+    const firstMethodInvocation = ctx.primarySuffix
+      ?.map(suffix => suffix.children.methodInvocationSuffix?.[0].children)
+      .find(methodInvocationSuffix => methodInvocationSuffix);
+    const isCapitalizedIdentifier =
+      nextToLastIdentifier && /^\p{Lu}/u.test(nextToLastIdentifier);
+    const shouldBreakBeforeFirstMethodInvocation =
```

Comment:
In the end, I ended up with something like that, WDYT ?

```ts
    ...
    const { newExpression, nextToLastIdentifier, firstMethodInvocation } = extract(ctx);
    const isCapitalizedIdentifier = isCapitalized(nextToLastIdentifier);

    const shouldBreakBeforeFirstMethodInvocation =
      countMethodInvocation > 1 &&
      !(isCapitalizedIdentifier) &&
      firstMethodInvocation !== undefined;
    ...
```

PS: the name of the extract method should be changed :)

Reply:
I ended up solving this via individual functions. Let me know what you think.